src/Indicator.vala

```diff
@@ -197,6 +202,17 @@ public class Power.Indicator : Wingpanel.Indicator {
         string? primary_text = null;
         string? secondary_text = null;
         if (display_device != null) {
+            if (display_device.percentage == LOW_BATTERY_PERCENTAGE && !display_device.is_charging) {
+                is_showing_low_battery_percent = true;
+                display_widget.show_percentage ();
+            }
+
+            /* Hide low battery percentage after plug charger if user is not showing percentage */
+            var is_showing_percent = settings.get_boolean ("show-percentage");
+            if (is_showing_low_battery_percent && display_device.is_charging && !is_showing_percent) {
+                settings.set_boolean ("show-percentage", false);
```

Comment:
Don't change the user setting (for reasons previously discussed).  Use `display_widget.set_show_percentage (false)`